Assets/Scripts/Waypoint Scripts/Waypoint.cs

```diff
@@ -9,9 +9,8 @@ public class Waypoint
     {
         public Drone referenceDrone; // The drone whose path this waypoint belongs to
         public GameObject gameObjectPointer; // This is the related game object
-        public ArrayList ROSpoints; // Keeps track of the ROSpoints received over the ROSBridge
+        //public ArrayList ROSpoints; // Keeps track of the ROSpoints received over the ROSBridge
```

Comment:
after all of this, where is the waypoint location stored? inside the gameObjectPointer?

Reply:
It's declared in the Waypoint struct, and held by Waypoint GameObject instances. Each waypoint object therefore holds its own coordinates. The waypoints ArrayList holds all the Waypoint GameObjects.